ndk/src/native_window.rs

```diff
@@ -1,21 +1,47 @@
-//! Bindings for `ANativeWindow`
+//! Bindings for [`ffi::ANativeWindow`]
 use std::ptr::NonNull;
 
-#[derive(Clone, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
+#[derive(Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
 pub struct NativeWindow {
     ptr: NonNull<ffi::ANativeWindow>,
 }
 
 unsafe impl Send for NativeWindow {}
 unsafe impl Sync for NativeWindow {}
 
+impl Drop for NativeWindow {
+    fn drop(&mut self) {
+        unsafe { ffi::ANativeWindow_release(self.ptr.as_ptr()) }
+    }
+}
+
+impl Clone for NativeWindow {
+    fn clone(&self) -> Self {
+        unsafe {
+            ffi::ANativeWindow_acquire(self.ptr.as_ptr());
+            Self { ptr: self.ptr }
+        }
+    }
+}
+
 impl NativeWindow {
+    /// Assumes ownership of `ptr`
+    ///
     /// # Safety
-    /// `ptr` must be a valid pointer to an Android `ANativeWindow`.
+    /// `ptr` must be a valid pointer to an Android [`ffi::ANativeWindow`].
     pub unsafe fn from_ptr(ptr: NonNull<ffi::ANativeWindow>) -> Self {
         Self { ptr }
     }
 
+    /// Acquires ownership of `ptr`
+    ///
+    /// # Safety
+    /// `ptr` must be a valid pointer to an Android [`ffi::ANativeWindow`].
+    pub unsafe fn clone_from_ptr(ptr: NonNull<ffi::ANativeWindow>) -> Self {
```

Comment:
Naming suggestions welcome here. There are some cases (`fn from_ptr()`) where Android gives us a pointer that we own, ie. we should call `ANativeWindow_release` when done (and not call `ANativeWindow_acquire` when constructing our owned wrapper). There are also cases - `on_window_created` + `on_window_destroyed` - where Android does its own refcounting/lifetime management. If we wish to participate we should _also_ increment the refcount before even thinking about releasing it again.